set.mm

```diff
@@ -647669,29 +647853,497 @@ standardize vectors to a length (norm) of one, but such definitions require
+    $( A counterexample to FLT stays valid when scaled.  The hypotheses are
+       more general than they need to be for convenience.  (There does not seem
+       to be a standard term for Fermat or Pythagorean triples extended to any
+       ` N e. NN0 ` , hence the label is more about the context in which this
+       theorem is used).  (Contributed by SN, 20-Aug-2024.) $)
+    fltmul $p |- ( ph
+       -> ( ( ( S x. A ) ^ N ) + ( ( S x. B ) ^ N ) ) = ( ( S x. C ) ^ N ) ) $=
+      ( cexp co cmul caddc expcld adddid oveq2d mulexpd eqtr3d oveq12d 3eqtr4d
+      ) AEFMNZBFMNZONZUDCFMNZONZPNZUDDFMNZONZEBONFMNZECONFMNZPNEDONFMNAUDUEUGPN
+      ZONUIUKAUDUEUGAEFGKQABFHKQACFIKQRAUNUJUDOLSUAAULUFUMUHPAEBFGHKTAECFGIKTUB
+      AEDFGJKTUC $.
+  $}
+
+  ${
+    fltdiv.s $e |- ( ph -> S e. CC ) $.
+    fltdiv.0 $e |- ( ph -> S =/= 0 ) $.
+    fltdiv.a $e |- ( ph -> A e. CC ) $.
+    fltdiv.b $e |- ( ph -> B e. CC ) $.
+    fltdiv.c $e |- ( ph -> C e. CC ) $.
+    fltdiv.n $e |- ( ph -> N e. NN0 ) $.
+    fltdiv.1 $e |- ( ph -> ( ( A ^ N ) + ( B ^ N ) ) = ( C ^ N ) ) $.
+    $( A counterexample to FLT stays valid when scaled.  The hypotheses are
+       more general than they need to be for convenience.  (Contributed by SN,
+       20-Aug-2024.) $)
+    fltdiv $p |- ( ph
+       -> ( ( ( A / S ) ^ N ) + ( ( B / S ) ^ N ) ) = ( ( C / S ) ^ N ) ) $=
+      ( cexp co cdiv caddc expcld nn0zd expdivd expne0d divdird oveq12d 3eqtr4d
+      oveq1d eqtr3d ) ABFNOZEFNOZPOZCFNOZUHPOZQOZDFNOZUHPOZBEPOFNOZCEPOFNOZQODE
+      POFNOAUGUJQOZUHPOULUNAUGUJUHABFILRACFJLRAEFGLRAEFGHAFLSUAUBAUQUMUHPMUEUFA
+      UOUIUPUKQABEFIGHLTACEFJGHLTUCADEFKGHLTUD $.
+  $}
+
+  ${
+    flt0.a $e |- ( ph -> A e. CC ) $.
+    flt0.b $e |- ( ph -> B e. CC ) $.
+    flt0.c $e |- ( ph -> C e. CC ) $.
+    flt0.n $e |- ( ph -> N e. NN0 ) $.
+    flt0.1 $e |- ( ph -> ( ( A ^ N ) + ( B ^ N ) ) = ( C ^ N ) ) $.
+    $( A counterexample for FLT does not exist for ` N = 0 ` .  (Contributed by
+       SN, 20-Aug-2024.) $)
+    flt0 $p |- ( ph -> N e. NN ) $=
+      ( wcel cc0 wne cexp co caddc c1 exp0d wceq oveq2 cn0 cn c2 sn-1ne2 necomi
+      1p1e2 eqnetri a1i oveq12d 3netr4d eqeq12d syl5ibcom imp mteqand sylanbrc
+      elnnne0 ) AEUAKELMEUBKIAELBLNOZCLNOZPOZDLNOZAQQPOZQUSUTVAQMAVAUCQUFQUCUDU
+      EUGUHAUQQURQPABFRACGRUIADHRUJAELSZUSUTSZABENOZCENOZPOZDENOZSVBVCJVBVFUSVG
+      UTVBVDUQVEURPELBNTELCNTUIELDNTUKULUMUNEUPUO $.
+  $}
+
+  ${
+    fltdvdsabdvdsc.a $e |- ( ph -> A e. NN ) $.
+    fltdvdsabdvdsc.b $e |- ( ph -> B e. NN ) $.
+    fltdvdsabdvdsc.c $e |- ( ph -> C e. NN ) $.
+    fltdvdsabdvdsc.n $e |- ( ph -> N e. NN ) $.
+    fltdvdsabdvdsc.1 $e |- ( ph -> ( ( A ^ N ) + ( B ^ N ) ) = ( C ^ N ) ) $.
+    $( Any factor of both ` A ` and ` B ` also divides ` C ` .  This
+       establishes the validity of ~ fltabcoprmex .  (Contributed by SN,
+       21-Aug-2024.) $)
+    fltdvdsabdvdsc $p |- ( ph -> ( A gcd B ) || C ) $=
```

Comment:
As you proved in the theorem above, N can be in NN0. Also, C can be in ZZ, and maybe A and B too.

It would certainly be nice to be as general as possible, but it may be a bit annoying in the proofs. Hopefully the generalizatio of the earlier results to ZZ may reduce the needs to treat zero separately.

The same questions will appear repeatedly below. Since it is your mathbox, these are non-blocking remarks.

Reply:
Maybe it's possible (easier) to first keep the proof in `NN` as a kind of lemma, and then extend to `ZZ` in a subsequent proof?

Reply:
@tirix It may be. I cannot tell from here: it really depends on whether zero has to be treated separately in specific subproofs, or if there is no need for it because the above lemmas have been properly generalized.